### PR TITLE
chore: rename internal dtype functions

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: .github/workflows/publish_to_pypi.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
   publish-to-pypi:
     name: >-
       Publish Python 🐍 distribution 📦 to PyPI
-    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
+    if: startsWith(github.ref, 'refs/tags/v')  # only publish to PyPI on tag pushes
```
Comment (MarcoGorelli) [inline .github/workflows/publish_to_pypi.yml:33]: drive-by - tags should start with `v` if we're releasing

File: narwhals/dtypes.py [hunk 2]
```diff
@@ -85,51 +84,3 @@ class Enum(DType): ...
 
 
 class Date(TemporalType): ...
-
-
-def translate_dtype(plx: Any, dtype: DType) -> Any:
```
Comment (MarcoGorelli) [inline narwhals/dtypes.py:90]: turns out we didn't need this function 😄 😭

**Comment by FBruzzesi** [comment]
That's a nice cleanup indeed! How come that namespaces are not covered now? Aren't we using them somewhere? Or maybe we initialize them directly in other methods without passing through `__narwhals_namespace__`

**Comment by MarcoGorelli** [comment]
thanks!

> Or maybe we initialize them directly in other methods without passing through __narwhals_namespace__

i've done this in a couple of other places, but I think `Series.__narwhals_namespace__` is unreachable - might as well getrid of it for now then? can always add it back later if it helps somewhere

**Comment by FBruzzesi** [comment]
> i've done this in a couple of other places

Thanks! I think it makes complete sense!

> but I think `Series.__narwhals_namespace__` is unreachable - might as well getrid of it for now then? can always add it back later if it helps somewhere

I am very indifferent if keeping with no cover or removing it 😂

**Comment by MarcoGorelli** [comment]
thanks for your review!
